### PR TITLE
Turn on -spirv-allow-extra-diexpressions option

### PR DESCRIPTION
**Pull request by AlexeySotkin**
This option prevents crash in SPIRV translator if input IR contains DIExpression
operations which don't have mapping to OpenCL.DebugInfo.100 spec.

Signed-off-by: Alexey Sotkin <alexey.sotkin@intel.com>

### CODE REVIEW

File: clang/test/Driver/sycl-offload.c [hunk 1]
```diff
@@ -598,6 +598,10 @@
 // RUN: %clangxx %s -fsycl -fsycl-explicit-simd -### 2>&1 | FileCheck %s --check-prefix=CHK-FSYCL-ESIMD
 // CHK-FSYCL-ESIMD: llvm-spirv{{.*}}-spirv-allow-unknown-intrinsics
 
+/// Check "-spirv-allow-extra-diexpressions" option is emitted for llvm-spirv tool for debug mode
+// RUN: %clangxx %s -fsycl -g -### 2>&1 | FileCheck %s --check-prefix=CHK-FSYCL-DEBUG
+// CHK-FSYCL-DEBUG: llvm-spirv{{.*}}-spirv-allow-extra-diexpressions
+
```
Comment (AGindinson) [inline clang/test/Driver/sycl-offload.c:604]: Don't know if this check is useful, given that the option gets enabled for all cases.

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 2]
```diff
@@ -7834,6 +7834,9 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
   if (getToolChain().getTriple().isSYCLDeviceEnvironment()) {
     TranslatorArgs.push_back("-spirv-max-version=1.1");
     TranslatorArgs.push_back("-spirv-debug-info-version=legacy");
+    // Prevent crash in the translator if input IR contains DIExpression
+    // operations which don't have mapping to OpenCL.DebugInfo.100 spec.
+    TranslatorArgs.push_back("-spirv-allow-extra-diexpressions");
```
Comment (AGindinson) [inline clang/lib/Driver/ToolChains/Clang.cpp:7839]: I'd suggest safeguarding and adding similar code to `ToolChains/SYCL.cpp:SYCL::Linker::constructLLVMSpirvCommand`. @mdtoguchi, do you recall why it ended up with 2 similar Translator command creations? Was this part of @Fznamznon's effort on device code split?

File: clang/test/Driver/sycl-offload.c (hunk 1 above)
Comment (AlexeySotkin) [reply clang/test/Driver/sycl-offload.c]: Right. I have removed this check.

File: clang/lib/Driver/ToolChains/Clang.cpp (hunk 2 above)
Comment (AlexeySotkin) [reply clang/lib/Driver/ToolChains/Clang.cpp:7839]: I have added this option to `ToolChains/SYCL.cpp:SYCL::Linker::constructLLVMSpirvCommand`.

**Review by AGindinson — APPROVED** [review]
LGTM, thanks!

**Pull request merged**